ios/NativeBridge.swift

```diff
@@ -0,0 +1,181 @@
+                reject(NativeBridge.errorCode, error.errorDescription, error)
+            }
+        }
+            
+    }
+    
+    @objc public func webAuthLogout(scheme: String?, federated: Bool, resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) {
+        let builder = Auth0.webAuth(clientId: self.clientId, domain: self.domain)
+        builder.clearSession(federated: federated) { result in
+                switch result {
+                case .success:
+                    resolve(true)
+                case .failure(let error):
+                    reject(NativeBridge.errorCode, error.errorDescription, error)
+                }
+            }
+    }
+    
+    @objc public func saveCredentials(credentialsDict: [String: Any], resolve: RCTPromiseResolveBlock, reject: RCTPromiseRejectBlock) {
+        
+        guard let accessToken = credentialsDict[NativeBridge.accessTokenKey] as? String, let tokenType = credentialsDict[NativeBridge.tokenTypeKey] as? String, let idToken = credentialsDict[NativeBridge.idTokenKey] as? String else { reject(NativeBridge.errorCode, "Incomplete information provided for credentials", NSError.init(domain: NativeBridge.errorCode, code: -99999, userInfo: nil)); return; }
+        
+        let refreshToken = credentialsDict[NativeBridge.refreshTokenKey] as? String
+        let scope = credentialsDict[NativeBridge.scopeKey] as? String
+        var expiresIn: Date?
+         if let string = credentialsDict[NativeBridge.expiresInKey] as? String, let double = Double(string) {
+             expiresIn = Date(timeIntervalSinceNow: double)
+         } else if let double = credentialsDict[NativeBridge.expiresInKey] as? Double {
+             expiresIn = Date(timeIntervalSinceNow: double)
+         } else if let dateStr = credentialsDict[NativeBridge.expiresInKey] as? String {
+             let dateFormatter = DateFormatter()
+             dateFormatter.dateFormat = NativeBridge.dateFormat
+             expiresIn = dateFormatter.date(from: dateStr)
+         }
+     
+        
+        let credentials =  Credentials(
+            accessToken: accessToken,
+            tokenType: tokenType,
+            idToken:  idToken,
+            refreshToken: refreshToken,
+            expiresIn: expiresIn ?? Date(),
+            scope: scope,
+            recoveryCode: nil
+        )
+        resolve(credentialsManager.store(credentials: credentials))
+    }
+    
+    @objc public func getCredentials(scope: String?, minTTL: Int, parameters: [String: Any], forceRefresh: Bool, resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) {
+        if(forceRefresh) {
+            credentialsManager.renew { result in
+                switch result {
+                case .success(let credentials):
+                    resolve(credentials.asDictionary())
+                case .failure(let error):
+                    reject(NativeBridge.errorCode, error.errorDescription, error)
```

Comment:
Note that the Credentials Manager produces `CredentialsManagerError` error values, which have different cases, depending on what the error is: https://auth0.github.io/Auth0.swift/documentation/auth0/credentialsmanagererror

In the Flutter SDK we map the cases to unique error codes: https://github.com/auth0/auth0-flutter/blob/e9dfa49419a1047bfaaddaf7f752fbbed3c400e3/auth0_flutter/ios/Classes/CredentialsManager/CredentialsManagerExtensions.swift#L4